File: src/main/java/com/fructflow/fructflow/dbase/Delivery.java
package com.fructflow.fructflow.dbase;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


import java.time.LocalDate;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "deliveries")
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "supplier_id", nullable = false)
    private Supplier supplier;

    @Column(name = "delivery_date")
    private LocalDate deliveryDate;

    @JsonManagedReference
    @OneToMany(mappedBy = "delivery", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DeliveryItem> deliveryItems;
}

File: fructflow-frontend/src/types/Delivery.ts
import { Supplier } from './Supplier';
import { Product } from './Product';

export interface DeliveryItem {
    id?: number;
    product: Product; // теперь передаем полноценный объект продукта
    quantity: number;
    cost: number;
}

export interface Delivery {
    id?: number;
    supplier: Supplier; // теперь передаем объект поставщика
    deliveryDate: string;
    deliveryItems: DeliveryItem[];
}
